lib/activerecord-import/import.rb

```diff
@@ -996,7 +996,10 @@ def find_associated_objects_for_import(associated_objects_by_class, model)
 
         changed_objects = association.select { |a| a.new_record? || a.changed? }
         changed_objects.each do |child|
-          child.public_send("#{association_reflection.foreign_key}=", model.id)
+          Array(association_reflection.inverse_of&.foreign_key || association_reflection.foreign_key).each_with_index do |column, index|
+            child.public_send("#{column}=", Array(model.id)[index])
+          end
+
```

Comment:
Another note for the reviewers.

Let's now consider we want to recursively import the model CompositeBook and its children CompositeChapter.

[Here](https://github.com/fragkakis/activerecord-import/blob/rails_71_composite_pks/test/models/composite_book.rb#L12-L13) is the has_many relation on CompositeBook, mentioning the names of the columns id and author_id. These are the PK columns in the composite_books table.

As we are setting the ids in the children, we need the columns as they are called in the children table (composite_chapters), which is [here](https://github.com/fragkakis/activerecord-import/blob/rails_71_composite_pks/test/models/composite_chapter.rb#L5-L6). So we need the inverse relation (the belongs_to in CompositeChapter), which has the names composite_book_id and author_id.

In order not to break things, I kept the original association_reflection.foreign_key as a fallback.

Reply:
Looks good 👍 If an association doesn't explicitly state the `inverse_of` does AR infer that relationship so that this still works?